lib/p2p/Pool.ts

```diff
@@ -183,6 +183,25 @@ class Pool extends EventEmitter {
    * Attempt to create an outbound connection to a node using its known listening addresses.
    */
   private tryConnectNode = async (node: NodeConnectionInfo, retryConnecting = false) => {
+    if (!this.connectWithLastAddress(node)) {
+      this.connectWithAdvertisedAddresses(node, retryConnecting);
+    }
+  }
+
+  private connectWithLastAddress = async(node: NodeConnectionInfo) => {
```

Comment:
Can you put a space after `async`? That's the style throughout most of the codebase. It looks like there's a lint rule for this so I'll open a separate PR for it.

Reply:
Ah okay, currently there's no lint rule for it.

Reply:
Yup, but I just opened #497.